File: backend/api/src/main/java/com/kidslearning/api/model/Skill.java
package com.kidslearning.api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Skill {
    private String id;
    private Subject subject;
    private String unitCode;
    private String title;
    private String icon;
    private List<String> prerequisites;
    private List<String> itemIds;
}

File: mobile/lib/models/skill.dart
import 'subject.dart';

class Skill {
  final String id;
  final Subject subject;
  final String unitCode;
  final String title;
  final String icon;
  final List<String> prerequisites;
  final List<String> itemIds;

  Skill({
    required this.id,
    required this.subject,
    required this.unitCode,
    required this.title,
    this.icon = '',
    this.prerequisites = const [],
    this.itemIds = const [],
  });

  factory Skill.fromJson(Map<String, dynamic> json) {
    return Skill(
      id: json['id'] ?? '',
      subject: parseSubject(json['subject']),
      unitCode: json['unitCode'] ?? '',
      title: json['title'] ?? '',
      icon: json['icon'] ?? '',
      prerequisites: List<String>.from(json['prerequisites'] ?? []),
      itemIds: List<String>.from(json['itemIds'] ?? []),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'subject': subject.name,
        'unitCode': unitCode,
        'title': title,
        'icon': icon,
        'prerequisites': prerequisites,
        'itemIds': itemIds,
      };
}
